### DISCUSSION TITLE
Send a comment to PR or commit

### DISCUSSION THREAD

**Discussion by josemonsalve2** [jenkinsci/bitbucket-push-and-pull-request-plugin#215] [Ideas]
# Feature Request
Either as a build or post-build action, it would be nice to have a way to submit a comment into the PR or the commit. This should help expand the information provided in the bitbucket side. There should also be an interface that allows us users to send information from the build (e.g. through env variables) into the comment itself. I am not that familiarized with Jenkins to be able to propose a good solution. 

Here is the API that could be used for this
https://developer.atlassian.com/bitbucket/api/2/reference/resource/repositories/%7Busername%7D/%7Brepo_slug%7D/pullrequests/%7Bpull_request_id%7D/comments

**Comment by sdaconceicao** [comment]
Adding curl calls to the bitbucket api from a pipeline is fairly straightforward.  You can use jenkins credentials to pull in your auth against your bitbucket server. The code below does this against bitbucket server, bitbucket cloud has a different api but still can accomplish the same things.

`
environment {
    JENKINS_SVC = credentials('your-credential-id')
  }

 stage('Start Build') {
      steps {
        sh "curl -X POST https://your.bitbucket.com/rest/build-status/1.0/commits/${BITBUCKET_PULL_REQUEST_LATEST_COMMIT_FROM_SOURCE_BRANCH} -H 'Content-Type: application/json' -d '{\"state\":\"INPROGRESS\", \"key\":\"PR Build #${BUILD_NUMBER}\",  \"url\":\"https://your.jenkins.com/job/your-project/${BUILD_NUMBER}/console\" }' --user '${JENKINS_SVC}'"
      }
    }
}
post {
success {
      sh "curl -X POST https://your.bitbucket.com/rest/api/1.0/projects/GLU/repos/glu-docs-react/pull-requests/${BITBUCKET_PULL_REQUEST_ID}/comments -H 'Content-Type: application/json' -d '{\"text\":\"Build successful\"}' --user '${JENKINS_SVC}'"
      sh "curl -X POST https://your.bitbucket.com/rest/build-status/1.0/commits/${BITBUCKET_PULL_REQUEST_LATEST_COMMIT_FROM_SOURCE_BRANCH} -H 'Content-Type: application/json' -d '{\"state\":\"SUCCESSFUL\", \"key\":\"PR Build #${BUILD_NUMBER}\",  \"url\":\"https://your.jenkins.com/job/your-project/${BUILD_NUMBER}/console\" }' --user '${JENKINS_SVC}'"
      }
 
 failure {
      sh "curl -X POST https://your.bitbucket.com/rest/api/1.0/projects/GLU/repos/glu-docs-react/pull-requests/${BITBUCKET_PULL_REQUEST_ID}/comments -H 'Content-Type: application/json' -d '{\"text\":\"Build failed\"}' --user '${JENKINS_SVC}'"
      sh "curl -X POST https://your.bitbucket.com/rest/build-status/1.0/commits/${BITBUCKET_PULL_REQUEST_LATEST_COMMIT_FROM_SOURCE_BRANCH} -H 'Content-Type: application/json' -d '{\"state\":\"FAILED\", \"key\":\"PR Build #${BUILD_NUMBER}\",  \"url\":\"https://your.jenkins.com/job/your-project/${BUILD_NUMBER}/console\" }' --user '${JENKINS_SVC}'"
      }
      }
`
